File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/step/IframeStepDTO.java
package com.mokatest.platform.demos.domain.ui.dto.step;// iframe操作


import com.mokatest.platform.demos.domain.ui.uiEnum.iframe.SwitchIframeType;
import com.mokatest.platform.demos.domain.ui.dto.ElementDTO;
import lombok.Data;

@Data
public class IframeStepDTO extends StepBaseDTO {
    private String stepType;
    // 切换方式
    private SwitchIframeType switchIframeType;
    // 目标元素
    private ElementDTO element;
    private String url;
    private String iframeName;
    private Integer iframeIndex;
    private String iframeId;
}

File: frontEnd/src/types/dto/stepDTO/IframeStepDTO.ts
// iframe操作
import {StepBaseDTO} from "@/types/dto/base/StepBaseDTO";
import {SwitchIframeType} from "@/types/enum/iframe/SwitchIframeType";
import {ElementDTO} from "@/types/dto/ElementDTO";

export class IframeStepDTO extends StepBaseDTO {
    stepType: string = 'IFRAME';
    // 切换方式
    switchIframeType: SwitchIframeType | string = 'ID';
    // 目标元素
    element?: ElementDTO = new ElementDTO();
    url?: string = '';
    iframeName?: string = '';
    iframeIndex?: number = 0;
    iframeId?: string = '';
}






